### PR TITLE
Delete associated hasMany links when deleting records

### PR DESCRIPTION
**Pull request by tstirrat**
When a record is deleted, it will now traverse `belongsTo` relationships and remove the appropriate `hasMany` links.

e.g. when deleting user with id `abc`:

```js
groups: {
  -JftrTtQ00NA68dZZFZg: {
    name: 'group 1',
    users: {
      abc: true, // <-- this will be deleted
      def: true,
      xyz: true
    }
  }
}
```

#165 #143

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+fixed - Deleting a record now removes orphaned hasMany relationship links
```
Comment (jwngr) [inline changelog.txt]: Put backticks around `hasMany` so it shows up like code.

Comment (jwngr) [inline changelog.txt]: Also end each item with a period.

Comment (tstirrat) [inline changelog.txt]: done

**Comment by oskarrough** [comment]
Thank you!!

**Pull request merged**